src/@utils/downloadJSON.ts

```diff
@@ -0,0 +1,23 @@
+export function isValidJSON(json: any): boolean {
+  try {
+    JSON.parse(json)
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+// Download JSON file
+export function downloadJSON(json: any, filename: string): void {
+  // check if the json is valid
+  if (isValidJSON(JSON.stringify(json))) {
```

Comment:
issue(blocking): `JSON.stringify()` will always produce valid JSON, or run into an error itself. In this case the function should expect a JSON string as input rather than `any` object.